src/android/com/megster/cordova/BluetoothSerial.kt

```diff
@@ -26,27 +26,26 @@ class BluetoothSerial : CordovaPlugin() {
     override fun execute(action: String, args: CordovaArgs, callbackContext: CallbackContext): Boolean {
         LOG.d(TAG, "action = $action")
         var validAction = true
+
+        try {
```

Comment:
i guess - not a huge fan of this pattern, but not worth blocking

Reply:
correct, the address is unavailable when the bluetooth is disabled

Reply:
then you probably want to have an isEnabled check in here too

Reply:
it might not be a blocking call though so iono

Reply:
my thinking it was covered by the function `enableBluetoothIfNecessary`.  if unable to enable then it throws and that is caught and we exits before applying the action.  

```
    private fun enableBluetoothIfNecessary() {
        if (!bluetoothAdapter.isEnabled) {
            bluetoothAdapter.enable()
        }
    }
```

Reply:
truth - i guess im just saying you could have a call to validate it is enabled if you wanted to be super safe! :P  - not blocking - feel free to merge at will!

Reply:
i'll add a ticket to understand whether this is async and add a check etc.

Reply:
it looks like its blocking https://github.com/aosp-mirror/platform_frameworks_base/blob/nougat-release/core/java/android/bluetooth/BluetoothAdapter.java#L899

Reply:
it looks like it returns a bool we could leverage 

     * @return true to indicate adapter startup has begun, or false on
     *         immediate error